criu/arch/mips/include/asm/restorer.h

```diff
@@ -0,0 +1,60 @@
+#ifndef __CR_ASM_RESTORER_H__
+#define __CR_ASM_RESTORER_H__
+
+#include "asm/types.h"
+#include <compel/asm/fpu.h>
+#include "images/core.pb-c.h"
+#include <compel/plugins/std/syscall-codes.h>
+#include <compel/asm/sigframe.h>
+
+static inline void restore_tls(tls_t *ptls) { 
+	asm volatile(							
+		     "move $4, %0				    \n"	
+		     "li $2,  "__stringify(__NR_set_thread_area)"  \n" 
+		     "syscall					    \n"	
+		     :							
+		     : "r"(*ptls)					
+		     : "$4","$2","memory");
+}
+static inline int arch_compat_rt_sigaction(void *stack, int sig, void *act)
+{
+	return -1;
+}
+static inline int set_compat_robust_list(uint32_t head_ptr, uint32_t len)
+{
+	return -1;
+}
+
+#define RUN_CLONE_RESTORE_FN(ret, clone_flags, new_sp, parent_tid,      \
+			     thread_args, clone_restore_fn)      \
+	asm volatile(							\
+		     "move $5, %6				\n"	\
+		     "move $6, %5				\n"	\
+		     "move $7, %1				\n"	\
+		     "move $8, %3				\n"	\
+		     "move $9, %4				\n"	\
+		     "break  					\n"	\
+		     : "=r"(ret)					\
+		     : "r"(clone_flags),				\
+		       "m"(new_sp),					\
+		       "r"(&parent_tid),				\
+		       "r"(&thread_args[i].pid),			\
+		       "r"(clone_restore_fn),				\
+		       "r"(&thread_args[i])				\
+		     :  "memory")
+
+#define RUN_CLONE3_RESTORE_FN(ret, clone_args, size, args, \
+			      clone_restore_fn)	do { \
+	pr_err("This architecture does not support clone3() with set_tid, yet!\n"); \
+	ret = -1; \
```

Comment:
And doing it for clone3() would also be nice. I was to test the clone3() code on all other architectures natively, but I have no access to mips64.

Reply:
I am working on kernel version v4.19 for mips architecture.  so I can`t check clone3().